src/bgen/Extensions/PlatformNameExtensions.cs

```diff
@@ -1,132 +1,74 @@
-using System;
-using System.IO;
-using ObjCRuntime;
-using Xamarin.Utils;
+using System.Diagnostics.CodeAnalysis;
 
 public static class PlatformNameExtensions {
 
-	public static string GetApplicationClassName (this PlatformName currentPlatform)
+	public static bool TryGetApplicationClassName (this PlatformName currentPlatform, [NotNullWhen (true)] out string? className)
 	{
 		switch (currentPlatform) {
 		case PlatformName.iOS:
 		case PlatformName.WatchOS:
 		case PlatformName.TvOS:
 		case PlatformName.MacCatalyst:
-			return "UIApplication";
+			className = "UIApplication";
```

Comment:
Question - would these ever have to be fully qualified?

Reply:
This is just a map, that will happen later in the code emitter. This class only provides helper methods that return the mapping.